visualization-project-laptop-defects/visualization_project_laptop_defects.py

```diff
@@ -0,0 +1,105 @@
+# -*- coding: utf-8 -*-
+"""visualization-project-laptop-defects.ipynb
+
+Automatically generated by Colab.
+
+Original file is located at
+    https://colab.research.google.com/drive/1luR1EyBWfxAuTP79aon700eziKW2caQX
+"""
+
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+
+file_path = '/content/Data_Analysis_Project.xlsx'
+df = pd.read_excel(file_path, sheet_name='Data')
+df
+
+# 1
+df['Month ID(YYYYMM)'] = pd.to_datetime(df['Month ID(YYYYMM)'].astype(str), format='%Y%m')
+
+# Set the datetime column as the index
+df.set_index('Month ID(YYYYMM)', inplace=True)
+
+# Plot the data
+plt.figure(figsize=(10, 6))
+plt.plot(df.index, df['Defects'], marker='o')
+
+# Format the x-axis to show month abbreviations
+plt.gca().xaxis.set_major_formatter(plt.matplotlib.dates.DateFormatter('%b %Y'))
+
+plt.title('Defects Over Time')
+plt.xlabel('Month')
+plt.ylabel('Defects')
+plt.grid(True)
+plt.show()
+
+# 2
+plt.figure(figsize=(10, 6))
+plt.plot(df.index, df['Opportunities'], marker='o')
+
+# Format the x-axis to show month abbreviations
+plt.gca().xaxis.set_major_formatter(plt.matplotlib.dates.DateFormatter('%b %Y'))
+
+plt.title('Opportunities Over Time')
+plt.xlabel('Month')
+plt.ylabel('Opportunities')
+plt.grid(True)
+plt.show()
+
+# 3
+
+defect_rate = df['Defects']/df['Opportunities']
+mean_defect_rate = defect_rate.mean()
+limit_2std = df['2 Sigma limit']/1000000
+limit_3std = df['3 Sigma limit']/1000000
+std_defect_rate = defect_rate.std()
+
+plt.figure(figsize=(10, 6))
+plt.plot(df.index, defect_rate, marker='o', linestyle='-', color='r', label='Defect Rate')
+plt.axhline(y=mean_defect_rate, color='blue', linestyle='--', label='Mean Defect Rate')
+plt.axhline(y=limit_2std[0], color='orange', linestyle='--', label='UCL (Mean + 2σ)')
+plt.axhline(y=limit_3std[0], color='green', linestyle='--', label='UCl (Mean + 3σ)')
+plt.title('Defect Rate Over Time with Control Limits')
+plt.xlabel('Time')
+plt.ylabel('Defect Rate')
+plt.legend()
+plt.gca().xaxis.set_major_formatter(plt.matplotlib.dates.DateFormatter('%b %Y'))
+
+# 4
```

Comment:
The first (left) y-axis should be for the number of defects/opportunities and the second (right) y-axis should be for the defect rate. Also make sure to include all variables in the legend. I have attached an image of what the final plot should look like.
![opp_vs_def](https://github.com/yhernandez55/ds-m1-0424-code-solutions/assets/19630777/6a1aa7c1-b98d-4f81-a8ff-2dc76fa5cf5e)